addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1341,6 +1355,111 @@ const DatetimePickerUserValueWidget = InputUserValueWidget.extend({
+        const order = this.el.dataset.order ? this.el.dataset.order : 'name asc';
+        const limit = this.el.limit ? parseInt(this.el.limit) : 5;
+
+        $.widget("custom.m2oAutocomplete", $.ui.autocomplete, {
+            _renderItem: (ul, item) => {
+                const li = $(`<li><div>${item.label}</div></li>`);
+                if (item.value.city || item.value.country_id) {
+                    li.find('div').append(`<small class="text-muted">(${item.value.city} ${item.value.country_id && item.value.country_id[1]})</small>`);
+                }
+                return li.appendTo(ul);
+            },
+        });
+        $(this.inputEl).m2oAutocomplete({
+            source: (request, response) => {
+                const search = request.term;
+                if (!search || !search.length) {
+                    return;
+                }
+                return this._rpc({
+                    route: '/web_editor/name_search_read',
+                    params: {
+                        model: model,
+                        name: search,
+                        limit: limit,
+                        fields: fields,
+                        order: order,
+                    }
+                }).then((result) => {
+                    const res = result.map(el => ({
+                        label: el.display_name,
+                        value: el,
+                    }));
+                    response(res);
+                });
+            },
+            focus: (ev, ui) => {
+                ev.preventDefault();
+            },
+            select: (ev, ui) => {
+                ev.preventDefault();
+                this._value = `${ui.item.value.id}`;
+                this._methodsParams.widgetPossibleValues.push({
+                    id: this._value,
+                    value: ui.item.value,
+                });
+                this.inputEl.value = '';
+                this.inputEl.blur();
+                this.getParent().close();
+                this.notifyValueChange(false);
+            },
+        });
+    },
+
+    /**
+     * @override
+     */
+    getTextValue: function () {
+        const value = this._methodsParams.widgetPossibleValues.find(el => el.id === this._value).value;
+        return value && value.display_name;
+    },
+    /**
+     * @override
+     */
+    loadMethodsData: function () {
+        this._super(...arguments);
+        const currentRecord = this.el.dataset.currentRecord;
+        if (currentRecord) {
+            this._methodsParams.widgetPossibleValues.push(JSON.parse(currentRecord));
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+    /**
+     * @override
+     */
+    _updateUI: async function () {
+        const value = this.inputEl.value;
+        await this._super(...arguments);
+        this.inputEl.value = value;
```

Comment:
This just cancels the updateUI of the InputUserValueWidget. Can I bypass super and call the updateUI of UserValueWidget ?

Reply:
Probably best to do like you did here.